document_merge_service/api/management/commands/clean_dangling_objects.py

```diff
@@ -0,0 +1,82 @@
+import boto3
+import botocore
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from document_merge_service.api.models import Template
+
+
+S3_ENDPOINT_URL = settings.S3_STORAGE_OPTIONS["endpoint_url"]
+S3_ACCESS_KEY_ID = settings.S3_STORAGE_OPTIONS["access_key"]
+S3_SECRET_ACCESS_KEY = settings.S3_STORAGE_OPTIONS["secret_key"]
+S3_BUCKET = settings.S3_STORAGE_OPTIONS["bucket_name"]
+
+
+class Command(BaseCommand):
+    help = "Cleans up unreferenced files and warns about unencrypted files in the storage bucket"
+
+    def get_object_keys(self, s3_client):
+        keys = []
+        have_next = True
+        continuation_token = None
+        while have_next:
+            data = s3_client.list_objects_v2(
+                Bucket=S3_BUCKET,
+                **(
+                    {"ContinuationToken": continuation_token}
+                    if continuation_token
+                    else {}
+                ),
+            )
+            if data["KeyCount"] > 0:
+                keys += [o["Key"] for o in data["Contents"]]
+            have_next = data["IsTruncated"]
+            if have_next:
+                continuation_token = data["NextContinuationToken"]
+        return keys
+
+    def delete_file(self, s3_client, stats: dict, fname: str):
+        print(f"Deleting unreferenced file: {fname}")
+        # TODO: delete for real
+        # s3_client.delete_object(Bucket=S3_BUCKET, Key=fname)
```

Comment:
This should be uncommented before merge.

Reply:
As discussed: Lets add a flag like `--commit` instead, and do the dry run by default.

Reply:
Added a `--commit` option. It now does a dry-run by default unless the option is passed.